### ISSUE TITLE
Dependency tracking is not collecting HTTP requests made with new SocketsHttpHandler

### ISSUE THREAD

**Issue by marnilss** [microsoft/ApplicationInsights-dotnet#1434]
If you are reporting bug/issue, please provide detailed Repro instructions.

## Repro Steps
1. Create project, e.g. Web API, with ASP.NET Core e.g 2.2
2. Make requests from e.g. a ApiController with HttpClient - make sure to create/use a SocketsHttpHandler instead of standard HttpHandler.
3. Enable Application Insights in project.
4. Run project and trigger your ApiController so it makes requests using the HttpClient defined in step 2.

##  Actual Behavior
Your HttpClient requests, from the ApiController, does NOT appear in the _dependencies_ table.

## Expected Behavior
Your HttpClient requests, from the ApiController, are logged in the _dependencies_ table with URL path etc.

If you switch to use the old HttpClientHandler, instead of the new SocketsHttpHandler, the _dependencies_ table are populated with values as expected.

## Version Info 
SDK Version : 2.5.1
.NET Version :	 .NET Core 2.2				
How Application was onboarded with SDK(VisualStudio/StatusMonitor/Azure Extension) : Visual Studio and Nuget package 2.5.1
OS : Windows 10
Hosting Info (IIS/Azure WebApps/ etc) : IIS Express (running locally)

**Comment by davidsh** [comment]
The issue is being tracked here: https://github.com/dotnet/runtime/issues/31261.

**Comment by Hjortsberg** [comment]
This just bit me big time after implementing SocketsHttpHandler according to [this Microsoft blog post](https://devblogs.microsoft.com/cosmosdb/httpclientfactory-cosmos-db-net-sdk/) by Cosmos DB Engineer Matias Quaranta (ealsur). The reason we implemented SocketsHttpHandler was to mitigate TCP-exhaustion in an AppService that reaches out to a fair amount of sub systems. 
Matias recommends instantiation of a SocketsHttpHandler in his post whereas @wanjos above describes why this is not a good idea.

tldr;
From what information I can piece together, I can't currently have an application reach out to a lot of HTTP-dependencies without  socket exhaustion _and_ track the telemetry for the calls? 
Or am I missing some obvious solution, I'm kind of new the whole .Net universe so any pointers are greatly appreciated! :)
